state/secrets.go

```diff
@@ -905,6 +965,139 @@ func (st *State) WatchConsumedSecretsChanges(consumer string) StringsWatcher {
 	})
 }
 
+// WatchObsoleteRevisions returns a watcher for notifying
+// when secret revisions no longer have any consumers wanting
+// to use them.
+func (st *State) WatchObsoleteRevisions(owner string) StringsWatcher {
+	return newCollectionWatcher(st, colWCfg{
+		col: secretRevisionsC,
+		filter: func(key interface{}) bool {
+			secretMetadataCollection, closer := st.db().GetCollection(secretMetadataC)
+			defer closer()
+
+			secretRevisionsCollection, closer := st.db().GetCollection(secretRevisionsC)
+			defer closer()
+
+			var doc secretRevisionDoc
+			err := secretRevisionsCollection.FindId(key).One(&doc)
+			if err != nil {
+				return false
```

Comment:
should we handle errors better here?

Reply:
these methods do not have error handling - they were designed to only allow filtering on the id value itself.
without rewriting the watcher code, we just assume no event is fired on error